api/admin/controller/__init__.py

```diff
@@ -2539,3 +2545,97 @@ def sitewide_registration_document(self):
         # Sign a JWT with the private key to prove ownership of the site.
         token = jwt.encode(payload, private_key, algorithm="RS256")
         return dict(url=public_key_url, jwt=token)
+
+
+class AdminSearchController(AdminController):
+    """APIs for the admin search pages
+    Eg. Lists Creation
+    """
+
+    def search_field_values(self) -> dict:
+        """Enumerate the possible values for the search fields with counts
+        - Audience
+        - Distributor
+        - Genre
+        - Language
+        - Publisher
+        - Subject
+        """
+        library = flask.request.library
+        collection_ids = [coll.id for coll in library.collections]
+        return self._search_field_values_cached(collection_ids)
+
+    # 1 hour in-memory cache
+    @memoize(ttls=3600)
+    def _search_field_values_cached(self, collection_ids: List[int]) -> dict:
+        def _unzip(values: List[tuple]) -> dict:
+            """Covert a list of tuples to a {value0: value1} dictionary"""
+            return {a[0]: a[1] for a in values if type(a[0]) is str}
+
+        # Reusable queries
+        classification_query = (
+            self._db.query(Classification)
+            .join(Classification.subject)
+            .join(
+                LicensePool, LicensePool.identifier_id == Classification.identifier_id
+            )
+            .filter(LicensePool.collection_id.in_(collection_ids))
+        )
+
+        editions_query = (
+            self._db.query(LicensePool)
+            .join(LicensePool.presentation_edition)
+            .filter(LicensePool.collection_id.in_(collection_ids))
+        )
+
+        # Concrete values
+        subjects = list(
```

Comment:
I'm curious do you know how will this work on frontend? On my local setup I had more than 1k items in this list, I'm not sure dropdown with 1k items is an option 😂

Reply:
The UI can manage that through searchable dropdowns, I've added the "counts" of works with each subject so that the UI can choose to show relevant results.